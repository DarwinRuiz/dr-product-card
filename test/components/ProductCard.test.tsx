
import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import { ProductCard, } from '../../src/components/ProductCard'
import { product1 } from '../data/products'

describe('ProductCard', () => {
    test('renders ProductCard component succeded', () => {
        const { container } = render(<ProductCard product={product1}>
            {
                () => (
                    <h1>Product Card</h1>
                )
            }
        </ProductCard>)

        expect(container).toMatchSnapshot()
    });


    test('should increment the counter', () => {
        const wrapper = render(<ProductCard product={product1}>
            {
                ({ count, incrementBy }) => (
                    <>
                        <h1>Product Card</h1>
                        <span title="count">{count}</span>
                        <button onClick={() => incrementBy(1)}>+</button>
                    </>
                )
            }
        </ProductCard>)

        const { container } = wrapper

        expect(container).toMatchSnapshot()

        const elementButton = wrapper.getByText('+')
        fireEvent.click(elementButton)

        const elementSpan = wrapper.getByTitle('count')
        expect(elementSpan.innerHTML).toBe('1')
    });
})