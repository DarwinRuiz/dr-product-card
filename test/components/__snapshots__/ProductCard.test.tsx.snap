// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders ProductCard component succeded 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
  </div>
</div>
`;

exports[`ProductCard should increment the counter 1`] = `
<div>
  <div
    class="productCard undefined"
  >
    <h1>
      Product Card
    </h1>
    <span
      title="count"
    >
      0
    </span>
    <button>
      +
    </button>
  </div>
</div>
`;
